src/main/scala/edu/ie3/powerFactory2psdm/converter/SwitchConverter.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.powerFactory2psdm.converter
+
+import edu.ie3.datamodel.models.input.NodeInput
+import edu.ie3.datamodel.models.input.connector.SwitchInput
+import edu.ie3.powerFactory2psdm.exception.pf.ConversionException
+import edu.ie3.powerFactory2psdm.model.powerfactory.Switch
+
+import java.util.UUID
+
+object SwitchConverter {
+
+  def convert(
+      rawSwitch: Switch,
+      nodeId2nodeInput: Map[String, NodeInput]
+  ): SwitchInput = {
+    val nodeA = nodeId2nodeInput.getOrElse(
+      rawSwitch.nodeAId,
+      throw ConversionException(
+        s"Can't find ${rawSwitch.nodeAId} for switch ${rawSwitch.id}"
+      )
+    )
+    val nodeB = nodeId2nodeInput.getOrElse(
+      rawSwitch.nodeBId,
+      throw ConversionException(
+        s"Can't find ${rawSwitch.nodeBId} for switch ${rawSwitch.id}"
+      )
+    )
+    val closed = rawSwitch.onOff match {
+      case 1d => true
+      case 0d => false
+      case _ =>
+        throw ConversionException(
+          s"The on/off parameter of switch ${rawSwitch.id} should be 1 or 0"
+        )
+    }
```

Comment:
You could decide on, if you'd like to leverage the check. Whenever there is an illegal argument given, the switch is considered to be open.

```suggestion
    val closed = rawSwitch.onOff == 1d
```

However, the check is very error prone anyways. If the attribute `onOff` is really double, there is a good chance, that you miss 1d, because the attribute is of value 1.0000000000003d. Please consider either using a different type or introduce a check, that considers a deviation on that.

Reply:
This should never be anything else other than 1 and 0 as PowerFactory only attributes these to values to the parameter. So in case this isn't the case I would rather want the program to crash instead of considering the switch as open to show that there is something wrong.

Reply:
But yes forgot about the floating point issues